### DISCUSSION TITLE
When using slickgrid-react, the sorters in the presets configuration work correctly on initial load. However, after updating the datasetHierarchical value via setState, the presets sorters no longer apply.

### DISCUSSION THREAD

**Discussion by hwlv** [ghiscoding/slickgrid-react#381] [General]
Initialize the SlickGrid component and set the presets with sorters.
Update the datasetHierarchical value.
Observe that the presets sorters no longer apply.

```tsx
<SlickGrid
  onCellChange={handleCellChange}
  loading={state.loading}
  ref={gridRef}
  gridId='base_info'
  gridOptions={state.gridOptions}
  columnDefinitions={columns}
  datasetHierarchical={state.list} />



const [state, setState] = useState({
  loading: false,
  gridOptions: {
    gridWidth: '100%',
    gridHeight: tableWrapHeight,
    autoEdit: true,
    autoCommitEdit: true,
    enableFiltering: true,
    enableTreeData: true,
    multiColumnSort: false,
    treeDataOptions: {
      columnId: 'name',
      childrenPropName: 'children',
      autoApproveParentItemWhenTreeColumnIsValid: true,
      autoRecalcTotalsOnFilterChange: false,
      excludeChildrenWhenFilteringTree: false,
    },
    presets: {
      sorters: [{ columnId: 'index', direction: 'asc' }]
    }
  },
  list: initialList
});

// Update datasetHierarchical
setState({
  list: updatedList
});
```

Expected Behavior:
The presets sorters should remain effective after updating the data.

Actual Behavior:
The presets sorters are no longer effective after the data is updated.

**Comment by ghiscoding** [comment]
First of all, I'm not a React developer. I maintain this project but it was created by another user a while ago, I just maintain it like all other SlickGrid projects that I have.

Second, Grid Presets are **only** effective on the first initial grid/page load. When the grid is loaded is the only time the Grid Presets will be executed and **never after**. So yes the presets is gone after you update the data, it's normal since you're no longer on the first load. It should be explained in the [Grid State & Presets](https://ghiscoding.gitbook.io/slickgrid-react/grid-functionalities/grid-state-preset) docs (again if it's not then please contribute improvements on the docs).

If you want to change Filters or Sorters (sorting) dynamically after the initial load, then you can't use grid presets. You need to use different functions to do that and all of that is shown in the multiple examples. The [Example 6](https://ghiscoding.github.io/slickgrid-react/#/example6) has demo of how to dynamically changing Filters/Sorting, which is probably what you're looking for.

Please read the documentation and look at all the demos before asking all these questions. Thanks

**Reply by hwlv** [reply]
Thank you so much for your detailed and prompt response! I really appreciate the clarification on Grid Presets and the documentation guidance. I’ll make sure to review the examples and documentation more thoroughly to address the dynamic changes in Filters and Sorters.

Thanks again for your help and for maintaining this project!

Best regards！

**Reply by hwlv** [reply]
I tried this, and it worked
  
  treeDataOptions: {
            initialSort: { columnId: 'index', direction: 'asc' }
        },
